src/main/Yardarm/Generation/Tag/TagTypeGeneratorBase.cs

```diff
@@ -29,6 +30,13 @@ protected TagTypeGeneratorBase(ILocatedOpenApiElement<OpenApiTag> tagElement, Ge
             string methodName = Context.NameFormatterSelector.GetFormatter(NameKind.AsyncMethod)
                 .Format(operation.Element.OperationId);
 
+            // This is here to show a useful error that there is an error with the Spec provided. If the operation
+            // is missing the name attribute in the OpenAPI spec, the MethodDeclaration step will fail
+            if (string.IsNullOrEmpty(methodName))
+            {
+                throw new NullReferenceException($"{nameof(GenerateOperationMethodHeader)} ran into an error. Please ensure the method at the path {operation.Key} {operation} is decorated correctly with the OperationId present.");
+            }
+
```

Comment:
There's actually a separate issue already open to address this problem, I'd prefer to leave it off this PR for the sake of the separation of issues.

```suggestion
```